test/sys/test_timerfd.rs

```diff
@@ -0,0 +1,49 @@
+use nix::errno::Errno;
+use nix::sys::time::{TimeSpec, TimeValLike};
+use nix::sys::timerfd::{ClockId, Expiration, TimerFd, TimerFlags, TimerSetTimeFlags};
+use nix::unistd::read;
+use nix::Error;
+use std::os::unix::io::AsRawFd;
+use std::time::Instant;
+
+#[test]
+pub fn test_timerfd_oneshot() {
+    let mut buffer = [0u8; 64];
+    let fd = TimerFd::new(ClockId::Monotonic, TimerFlags::empty()).unwrap();
+
+    let before = Instant::now();
+
+    fd.settime(
+        TimerSetTimeFlags::empty(),
+        Expiration::OneShot(TimeSpec::seconds(1)),
+    )
+    .unwrap();
+
+    read(fd.as_raw_fd(), &mut buffer).unwrap();
+
+    let millis = before.elapsed().as_millis();
+    assert!(millis < 1100 && millis > 900);
```

Comment:
You cannot assert that `millis < 1100`.  There is no guarantee that the timer will expire promptly, or that this process will run immediately after the timer expires.

Reply:
You are right, I removed the upper bound. Do you think setting a lower bound could be problematic in some corner cases ? I could not think of any "practical" problems with that.

Reply:
I don't know of any case where the lower bound could be a problem, unless the read gets interrupted by a signal.  What does Linux do in that case?  If it's possible for the read to return early with EINTR, then you should retry the read in your higher-level method.